### PR TITLE
Add a configuration file and a quiet mode

### PR DESCRIPTION
**Pull request by diml**
Once we don't stop at the first error, the output becomes messy. This PR adds display modes:

- `quiet`: only print errors
- `progress`: interactive display showing the number of rules processed so far and the number of parallel jobs running. This is the new default
- `short`: current default
- `verbose`: same as current `--verbose`

The display mode can be selected via the `--display` command line argument, or via the configuration file `~/.config/dune/config`.

TODO:
- [ ] Document display modes in the manual
- [ ] Document the configuration file in the manual

### CODE REVIEW

File: src/scheduler.ml [hunk 1]
```diff
@@ -79,34 +107,58 @@ let wait_for_process pid =
...
-    Fiber.Var.set info_var info
+    Fiber.Var.set t_var t
       (Fiber.with_error_handler (fun () -> fiber) ~on_error:Report_error.report)
   in
   Fiber.run
-    (Fiber.fork (fun () -> go_rec info) >>= fun _ ->
-     fiber)
+    (Fiber.fork_and_join_unit
+       (fun () -> go_rec t)
+       (fun () -> fiber))
```
Comment (rgrinberg) [inline src/scheduler.ml]: Previously, this change was simply omitted, right? I assume this is just for perf.

File: bin/main.ml [hunk 2]
```diff
@@ -85,6 +84,20 @@ module Main = struct
...
+
+  let create common =
+    Log.create ~display:common.config.display ()
+end
+
+module Scheduler = struct
+  include Jbuilder.Scheduler
+
+  let go ?log ~common fiber =
+    Scheduler.go ?log ~config:common.config fiber
```
Comment (rgrinberg) [inline bin/main.ml]: I wonder if we should just initialize the log here from `~common`. I think that's what we always do anywhere.

**Comment by rgrinberg** [comment]
Also it would be nice to enumerate the display options in `--help` as well. It's a bit of overkill to view the manual just for this.

**Comment by diml** [comment]
> Seems to work well for me and the new output is definitely an improvement. Other candidates for this config file are things like the diff cmd.

Indeed. We could also allow to customize the status line:

```scheme
(status_line "foo: ${todo}, jobs: ${jobs}")
```

One things I'm going to add straight after that is:

```scheme
(jobs 20)
```

> Also it would be nice to enumerate the display options in --help as well. It's a bit of overkill to view the manual just for this.

indeed

**Comment by diml** [comment]
I added the doc. I was finding that the man page was getting big, so I added `jbuilder help config`. I haven't updated the manual yet, though I find it painful to document things twice...

File: bin/main.ml (hunk 2 above)
Comment (diml) [reply bin/main.ml]: indeed, although I think there is a case where we call the scheduler twice

File: src/scheduler.ml (hunk 1 above)
Comment (diml) [reply src/scheduler.ml]: Yes, it was a mistake. This showed up after this patch as sometimes the status line wasn't cleaned up at the end

**Pull request merged**